File: apps/api/internal/stellar/operator.go
package stellar

import (
	"fmt"
	"os"
	"strings"

	"github.com/stellar/go-stellar-sdk/keypair"
)

// OperatorSecret returns STELLAR_{TESTNET|MAINNET}_OPERATOR_SECRET.
func OperatorSecret(environment string) string {
	suffix := "TESTNET"
	if environment == "production" {
		suffix = "MAINNET"
	}
	return strings.TrimSpace(os.Getenv("STELLAR_" + suffix + "_OPERATOR_SECRET"))
}

func IsOperatorConfigured(environment string) bool {
	return OperatorSecret(environment) != ""
}

func OperatorKeypair(environment string) (*keypair.Full, error) {
	secret := OperatorSecret(environment)
	if secret == "" {
		suffix := "TESTNET"
		if environment == "production" {
			suffix = "MAINNET"
		}
		return nil, fmt.Errorf("STELLAR_%s_OPERATOR_SECRET is not configured", suffix)
	}
	kp, err := keypair.ParseFull(secret)
	if err != nil {
		return nil, fmt.Errorf("invalid operator secret: %w", err)
	}
	return kp, nil
}

func OperatorPublicKey(environment string) (string, error) {
	kp, err := OperatorKeypair(environment)
	if err != nil {
		return "", err
	}
	return kp.Address(), nil
}

File: apps/web/src/lib/stellar/operator.ts
import { Keypair } from "@stellar/stellar-sdk";
import type { Organization } from "@/lib/db/schema";

function readEnv(name: string) {
  return process.env[name]?.trim() || null;
}

/**
 * Single Stellar operator keypair per network.
 * Used for Soroban authorization, classic escrow deposits, and sandbox simulation.
 */
export function getStellarOperatorSecret(
  environment: Organization["environment"]
) {
  const suffix = environment === "production" ? "MAINNET" : "TESTNET";
  return readEnv(`STELLAR_${suffix}_OPERATOR_SECRET`);
}

export function getStellarOperatorKeypair(environment: Organization["environment"]) {
  const secret = getStellarOperatorSecret(environment);

  if (!secret) {
    throw new Error(
      `STELLAR_${environment === "production" ? "MAINNET" : "TESTNET"}_OPERATOR_SECRET is not configured`
    );
  }

  return Keypair.fromSecret(secret);
}

export function getStellarOperatorPublicKey(
  environment: Organization["environment"]
) {
  return getStellarOperatorKeypair(environment).publicKey();
}

export function isStellarOperatorConfigured(
  environment: Organization["environment"]
) {
  return Boolean(getStellarOperatorSecret(environment));
}
